core/modules/core/src/configuration_environment.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 Klepsydra Technologies AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific lang
+
+#include <fstream>
+
+#include <cereal/archives/json.hpp>
+
+#include <klepsydra/core/configuration_environment.h>
+#include <klepsydra/core/configuration_properties.h>
+
+#include <spdlog/spdlog.h>
+
+namespace kpsr {
+
+ConfigurationEnvironment::ConfigurationEnvironment() {}
+
+ConfigurationEnvironment::ConfigurationEnvironment(const std::string &configFileName,
+                                                   const std::string &rootNode)
+{
+    ConfigurationEnvironment tempEnvironment;
+    std::ifstream inputFileStream(configFileName, std::ios::binary);
+    {
+        ::cereal::JSONInputArchive iarchive(inputFileStream);
+        tempEnvironment.serialize(iarchive);
+    }
+    copyFrom(tempEnvironment, rootNode);
+}
+
+ConfigurationEnvironment::~ConfigurationEnvironment() {}
+
+void ConfigurationEnvironment::printGetError(bool result, const std::string &key) const
+{
+    if (!result) {
+        spdlog::warn("Environment does not have the property {}. Value not loaded", key);
```

Comment:
A minor thing: do you know if spdlog is set to show in which method the message has been printed? As it is now, a person who sees this message may not know where it happened.

Reply:
Not in this case. This is a customer message, no need to tell the class / method failing.